cmd/podman/images/scp.go

```diff
@@ -0,0 +1,350 @@
+		rep, err := toRemote(dials[0], remoteFiles[0], scpOpts.Save.Output, "")
+		if err != nil {
+			return err
+		}
+		fmt.Println(rep)
+		err = os.Remove(scpOpts.Load.Input)
+		if err != nil {
+			return err
+		}
+	// TODO: Add podman remote support
+	default: // else native load
+		if scpOpts.Tag != "" {
+			return errors.Wrapf(define.ErrInvalidArg, "Renaming of an image is currently not supported")
+		}
+		scpOpts.Save.Format = "oci-archive"
+		abiErr := abiEng.Save(context.Background(), scpOpts.DestImageName, []string{}, scpOpts.Save) // save the image locally before loading it on remote, local, or ssh
+		if abiErr != nil {
+			errors.Wrapf(abiErr, "could not save image as specified")
+		}
+		rep, err := abiEng.Load(context.Background(), scpOpts.Load)
+		if err != nil {
+			return err
+		}
+		fmt.Println("Loaded images(s): " + strings.Join(rep.Names, ","))
+		err = os.Remove(scpOpts.Load.Input)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// toRemote takes a dialed ssh client, and image information. it connects to the client
+// and copies the saved image dir over to the remote host and then loads it onto the machine
+// returns a string containing output or an error
+func toRemote(dial *ssh.Client, remoteFile string, localFile string, tag string) (string, error) {
+	f, err := os.Open(localFile) // open image file to copy
+	if err != nil {
+		return "", err
+	}
+	defer f.Close()
+	n, err := scpD.CopyTo(dial, localFile, remoteFile)
+	if err != nil {
+		errOut := (strconv.Itoa(int(n)) + " Bytes copied before error")
+		return " ", errors.Wrapf(err, errOut)
+	}
+	run := ""
+	if tag != "" {
+		return "", errors.Wrapf(define.ErrInvalidArg, "Renaming of an image is currently not supported")
+	}
+	podman := os.Args[0]
+	run = podman + " image load --input=" + remoteFile + ";rm " + remoteFile // run ssh image load of the file copied via scp
+	out, err := connection.ExecRemoteCommand(dial, run)
+	if err != nil {
+		return "", err
+	}
+	err = dial.Close()
+	if err != nil {
+		return "", err
+	}
+	return strings.TrimSuffix(out, "\n"), nil
+}
+
+// toRemote takes a dialed ssh client and image information. it connects to the client
+// and saves the specified image on the remote machine and then copies it to the specified local location
+// returns an error if one occurs.
+func fromRemote(dial *ssh.Client, image, remoteFile string, localFile string, tag string) error {
+	if tag != "" {
+		return errors.Wrapf(define.ErrInvalidArg, "Renaming of an image is currently not supported")
+	}
+	podman := os.Args[0]
+	run := podman + " image save " + image + " --format=oci-archive --output=" + remoteFile // run ssh image load of the file copied via scp. Files are reverse in thie case...
+	_, err := connection.ExecRemoteCommand(dial, run)
+	if err != nil {
+		return nil
+	}
+	n, err := scpD.CopyFrom(dial, remoteFile, localFile)
+	if err != nil {
+		errOut := (strconv.Itoa(int(n)) + " Bytes copied before error")
+		return errors.Wrapf(err, errOut)
+	}
```

Comment:
```go
ExecRemoteCommand(…, "rm " + remoteFile)
```
is now easy to do here.

(BTW it would probably be ideal to allocate the remoteFile within this function, see elsewhere, and immediately `defer` its removal.)